### PR TITLE
[SYCL] Add support for assert in windows NVPTX

### PR DESCRIPTION
**Pull request by jchlanda**
Until cxx standard library for NVPTX extension is implemented `_wassert` is provided
by libclc, that passes it to a call to `__assertfail`.

Fixes https://github.com/intel/llvm/issues/5922

### CODE REVIEW

File: clang/test/SemaSYCL/Inputs/assert.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#ifndef MOCK_ASSERT_H
+#define MOCK_ASSERT_H
+
+#define _CRT_WIDE_(s) L##s
```
Comment (smanna12) [inline clang/test/SemaSYCL/Inputs/assert.h:4]: Same comment here

File: clang/test/SemaSYCL/kernel-assert.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// REQUIRES: system-windows
+// RUN: %clang_cc1 -fsycl-is-device -triple nvptx64-nvidia-cuda -internal-isystem %S/Inputs/ -fsyntax-only -verify %s
+
+// expected-no-diagnostics
```
[the remark this answers is not in the document]
Comment (jchlanda) [reply clang/test/SemaSYCL/kernel-assert.cpp:4]: Done

File: clang/test/SemaSYCL/Inputs/assert.h (hunk 1 above)
Comment (jchlanda) [reply clang/test/SemaSYCL/Inputs/assert.h:4]: And done.

File: clang/test/SemaSYCL/kernel-assert.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+#include <assert.h>
+
+// Make sure that SYCL kernels can correctly call asserts on Windows. We do not
+// expect to see any warnings or compilation errors. On Windows targeting NVPTX
+// the implementation of assert is provided by libclc.
+//
+// Please note, that the execution of the assert is tested in llvm-test-suite.
```
Comment (elizabethandrews) [inline clang/test/SemaSYCL/kernel-assert.cpp:12]: If this functionality is being tested in the test suite and we no longer have FE changes, I don't think we need FE tests. Can you explain why you added this test? I'm not sure if I am missing something.

**Comment by aelovikov-intel** [comment]
+1 for removing FE tests. I'd rather let it go without tests than with tests in unrelated place.

**Review by elizabethandrews — APPROVED** [review]

**Review by pvchupin — APPROVED** [review]

**Pull request merged**